### PR TITLE
Incoming Peering Management

### PR DESCRIPTION
**Pull request by aleoli**
# Description

This pr allows a cluster to explicitly allow/deny an incoming peering from a remote cluster

* a new field has been added in the ClusterConfig to set the default behavior with the newly discovered clusters
* a new field has been added in the ForeignCluster to override the default behavior and to set the incoming policy for a specific remote cluster

# How Has This Been Tested?

- [x] locally on KinD
- [x] add unit test

### CODE REVIEW

File: internal/discovery/foreign-cluster-operator/foreign-cluster-controller.go [hunk 1]
```diff
@@ -415,12 +418,24 @@ func getPeeringPhase(foreignCluster *discoveryv1alpha1.ForeignCluster,
 		resourceRequest := &resourceRequestList.Items[0]
 		desiredDelete := !resourceRequest.Spec.WithdrawalTimestamp.IsZero()
 		deleted := !resourceRequest.Status.OfferWithdrawalTimestamp.IsZero()
-		if desiredDelete || deleted {
-			return discoveryv1alpha1.PeeringConditionStatusDisconnecting, resourceRequestDeletingReason,
-				fmt.Sprintf(resourceRequestDeletingMessage, foreignCluster.Status.TenantNamespace.Local), nil
+		offerState := resourceRequest.Status.OfferState
+
+		switch offerState {
```
Comment (palexster) [inline internal/discovery/foreign-cluster-operator/foreign-cluster-controller.go:423]: Some comments would be helpful here to understand each different case

**Review by palexster — APPROVED** [review]

**Review by alacuku — APPROVED** [review]

**Pull request merged**